src/plugins/data/common/index_patterns/index_patterns/index_patterns.ts

```diff
@@ -249,7 +250,9 @@ export class IndexPatternsService {
     try {
       const fields = await this.getFieldsForIndexPattern(indexPattern);
       const scripted = indexPattern.getScriptedFields().map((field) => field.spec);
-      indexPattern.fields.replaceAll([...fields, ...scripted]);
+      const fieldAttrs = indexPattern.getFieldAttrs();
+      const fieldsWithSavedAttrs = Object.values(this.fieldArrayToMap(fields, fieldAttrs));
```

Comment:
I think the fieldAttrs need to be applied to the scripted fields just as they are regular fields.

Reply:
will have a look

Reply:
```
const fieldsWithSavedAttrs = Object.values(this.fieldArrayToMap([...fieldsWithSavedAttrs, ...scripted], fieldAttrs));
indexPattern.fields.replaceAll(fieldsWithSavedAttrs);
```

Reply:
seems you already took care of it, many thx!

Reply:
@kertal I haven't, although the confusion is understandable considering my clumsy efforts to remove commented code yesterday.

Reply:
So now I have a look 😄

Reply:
@mattkime Maybe I miss the forest through the trees, but I think this is done in?
Line 255 

```
 indexPattern.fields.replaceAll([...fieldsWithSavedAttrs, ...scripted]);
```

Reply:
ah now I get it :) trees, trees, trees

Reply:
Trees in scripted fields: <img width="205" alt="Bildschirmfoto 2020-11-11 um 16 12 16" src="https://user-images.githubusercontent.com/463851/98828974-fad58d80-2438-11eb-9867-98386be2e167.png">

Reply:
thx for saving scripted fields!